src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -0,0 +1,185 @@
+
+                    if (!Settings.getSettingsFor(project).isBound && !Settings.getGlobalSettings().isPromotionDisabled && wasAutoAnalyzed) {
+                        processFullProjectPromotion(notifications)
+                    }
+
+                    if (nonReportAnalysisTriggers.contains(triggerType) && !wasAutoAnalyzed) {
+                        PropertiesComponent.getInstance().setValue(FIRST_AUTO_ANALYSIS_DATE, Instant.now().toEpochMilli().toString())
+                    }
+
+                    if (reportAnalysisTriggers.contains(triggerType)) {
+                        PropertiesComponent.getInstance().setValue(WAS_REPORT_EVER_USED, true)
+                    }
+                }
+            })
+        }
+    }
+
+    private fun getSonarLintProjectNotifications(source: FileEditorManager): SonarLintProjectNotifications {
+        return SonarLintUtils.getService(
+            source.project,
+            SonarLintProjectNotifications::class.java
+        )
+    }
+
+    private fun getSonarLintProjectNotifications(): SonarLintProjectNotifications {
+        return SonarLintUtils.getService(
+            project,
+            SonarLintProjectNotifications::class.java
+        )
+    }
+
+    private fun processFullProjectPromotion(notifications: SonarLintProjectNotifications) {
+        val firstAutoAnalysis = Instant.ofEpochMilli(PropertiesComponent.getInstance().getLong(FIRST_AUTO_ANALYSIS_DATE, 0L))
+        val wasReportEverUsed = PropertiesComponent.getInstance().getBoolean(WAS_REPORT_EVER_USED, false)
+
+        if (!wasReportEverUsed && firstAutoAnalysis.isBefore(Instant.now().minus(FULL_PROJECT_PROMOTION_DURATION))) {
+            UiUtils.Companion.runOnUiThread(project, Runnable {
+                SonarLintUtils.getService(project, PromotionProvider::class.java)
+                    .showPromotion(notifications, "Detect issues in your whole project")
+            })
+        }
+    }
+
+    private fun processAdvancedLanguagePromotion(notifications: SonarLintProjectNotifications, extension: String) {
+        val language = findLanguage(extension, languagesHavingAdvancedRules)
+
+        if (language != null) {
+            showPromotion(notifications, "Detect more security issues in your " + language.label + " files")
+        }
+    }
+
+    private fun processExtraLanguagePromotion(notifications: SonarLintProjectNotifications, extension: String) {
+        val language = findLanguage(extension, extraEnabledLanguagesInConnectedMode)
+
+        if (language != null) {
+            showPromotion(notifications, "Enable " + language.label + " analysis by connecting your project")
+        }
+    }
+
+    private fun findLanguage(extension: String, languages: Set<Language>): Language? {
+        return languages.find {
+            it.defaultFileSuffixes.any { suffix ->
+                suffix.equals(extension) || suffix.equals(".$extension")
```

Comment:
You can replace the `.equals` by `==`

Reply:
I prefer .equals

Reply:
IMO we should switch to `=`= because it raises a Sonar rule, so I'd change it because we eat our own dog food (unless it really does not make sense), and then because in standalone mode the issue will be raised again 😅

Reply:
The reason I prefer using .equals is that some languages have this == for comparing the value(Kotlin) and some languages (Java) have this for comparing the reference. In our codebase we use both languages, and if are in the habit of using == instead of .equals there is a chance that will use it thinking it compares the values of the strings but in fact it would be comparing the references

Reply:
IMO we should disable this rule